sdr.c

```diff
@@ -128,6 +131,9 @@ static sdr_handler sdr_handlers[] = {
 #ifdef ENABLE_LIMESDR
     { "limesdr", SDR_LIMESDR, limesdrInitConfig, limesdrShowHelp, limesdrHandleOption, limesdrOpen, limesdrRun, noStop, limesdrClose, noGetGain, noGetMaxGain, noGetGainDb, noSetGain },
 #endif
+#ifdef ENABLE_SOAPY
+    { "soapy", SDR_SOAPY, soapyInitConfig, soapyShowHelp, soapyHandleOption, soapyOpen, soapyRun, noStop, soapyClose, noGetGain, noGetMaxGain, noGetGainDb, noSetGain },
+#endif
```

Comment:
It would be nice to have the gain API implemented here

Reply:
I've made an attempt at implementing this but I'm not sure I understand how to handle stepping for devices like RTLSDR that has 3dB stepping?  My device seems happy with 1dB stepping so my implementation works for me but I realize it may fail for 3dB steps.  My SDRPlay device has two gain elements: IFGR and RFGR.  When I call `SoapySDRDevice_setGain()` with a gain of 7dB, IFGR is set to 27dB because IFGR minimum is 20dB.  This can be seen by calling `SoapySDRDevice_getGainElement()` for `IFGR`.  I don't know what kind of gain elements (reported by `SoapySDRDevice_listGains()`) devices like RTLSDR has and how `SoapySDRDevice_setGain()` affects them.

I've also modified the output to report the values of gain elements so that someone with a RTLSDR or other device can start the program and get information on what SoapySDR has for their device.  For example, here's what I see for `--gain 7`:
```
soapy: hardware key is RSPdx
soapy: available antennas: Antenna A, Antenna B, Antenna C
soapy: gain set to 7.0dB, IFGR=27.0dB, RFGR=0.0dB
```

That last line is showing gain of 7 because I started with `--gain 7` and IFRGR is showing 27dB because of IFGRmin+gain = 20dB+7dB.  These are the numbers reported by `SoapySDRDevice_getGain()` and `SoapySDRDevice_getGainElement()` after calling `SoapySDRDevice_setGain()` with a value of 7.

Reply:
I had a detailed comment here but then github ate it :/

tl;dr: call `SoapySDRDevice_getGainRange()` and use the step size in the returned range structure. i.e. you will have a max gain step index of `(range.maximum - range.minimum) / range.step` and for gain index X the gain in dB is `range.minimum +X * range.step`. Beware of `step == 0`, you probably want to force the step size to e.g. 1dB in that case. Let the SoapySDR driver worry about allocating the requested gain between gain elements.

You might want options to (a) let the user control initial gain per gain element (a generalization of your `--rfgain` option) and (b) to make the gain control API control a specific gain element only, rather than all elements.

Reply:
I've changed `soapyGetMaxGain()` to
`(range.maximum - range.minimum) / (!range.step ? 1 : range.step)`
and `soapyGetGainDb()` to
`(range.minimum + step) * (!range.step ? 1 : range.step)`

Regarding your last paragraph about generalizing, you're proposing enhancing the  `--gain` parameter to accept something like `--gain IFGR:x,RFGR:y`?